pkg/master/controller.go

```diff
@@ -301,20 +306,35 @@ func (r *masterCountEndpointReconciler) ReconcileEndpoints(serviceName string, i
 			Addresses: []api.EndpointAddress{{IP: ip.String()}},
 			Ports:     endpointPorts,
 		}}
-		return r.endpointRegistry.UpdateEndpoints(ctx, e)
+		return r.endpointRegistry.UpdateEndpoints(ctxDefault, e)
+	}
+
+	// Determine master count.
+	masterCount := r.masterCount
+
+	ctxSystem := api.WithNamespace(api.NewContext(), api.NamespaceSystem)
+	m, err := r.configmapRegistry.GetConfigMap(ctxSystem, componentconfig.ApiServerConfigMapName)
+	if err == nil {
+		if str, ok := m.Data[componentconfig.ApiServerConfigMapKey]; ok {
+			var config componentconfig.ApiServerConfiguration
+			err := json.Unmarshal([]byte(str), &config)
```

Comment:
You should use the Decoder to unmarshal the JSON string you pull out of `m.Data` (which is formatted as the external type, e.g. something from `componentconfig/v1alpha1/types.go`) into the corresponding struct for the external type, and then use `api.Scheme.Convert` to convert that to the internal (e.g. from `componentconfig/types.go`) type. 

See https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/server.go#L243 for an example of how this is currently done for dynamic config.